### ISSUE TITLE
The macro name is the same as the std member

### ISSUE THREAD

**Issue by XY-ShadowK** [open-source-parsers/jsoncpp#1591]
**Describe the bug**
```c++
#if !defined(isfinite)
#include <float.h>
#define isfinite _finite
#endif
```
The macro name `isfinite` is the same as `std::isfinite`, which can cause compiler to incorrectly replace `std::isfinite` to `std::_finite`.

**To Reproduce**
Just use `std::isfinite` after the inclusion of `jsoncpp.cpp`.

**Expected behavior**

**Desktop (please complete the following information):**
 - OS: Windows 10 22H2 x64

**Additional context**
Please use `std::isfinite` to check values, or use `_isfinite` directly or use other macro name to circumvent std member names.

**Comment by BillyDonahue** [comment]
I agree that's messed up to use a colliding name. We could just use some other name.

> Just use std::isfinite after the inclusion of jsoncpp.cpp.

But why would you be including a .cpp file?
The macros are safe in the json_writer.cpp file if used as intended.

**Comment by XY-ShadowK** [comment]
> But why would you be including a .cpp file? The macros are safe in the json_writer.cpp file if used as intended.

I followed the guidelines of [Amalgamated- (Possibly-outdated)](https://github.com/open-source-parsers/jsoncpp/wiki/Amalgamated-(Possibly-outdated)) to import `jsoncpp`.
If I want to compile a cross-platform project, can I import it this way and build with my own project? Or do I need to follow the instructions of [Build-with-VS-2019-on-Windows](https://github.com/open-source-parsers/jsoncpp/wiki/Build-with-VS-2019-on-Windows) and [Building](https://github.com/open-source-parsers/jsoncpp/wiki/Building) to build and import separately?

**Comment by BillyDonahue** [comment]
I think you misunderstood the guidelines.
The amalgamation script produces a .h and a .cpp file.

You add these to your project.
This means you add the single .cpp file to your Makefile or whatever you're using.
It doesn't mean that you #include it into your own files. It's still a separate compilation.

**Comment by XY-ShadowK** [comment]
As below?
So can it be built cross-paltform with my project?
```c++
cl src/mySrc.cpp jsoncpp-1.9.5/dist/jsoncpp.cpp /I jsoncpp-1.9.5/dist
g++ src/mySrc.cpp jsoncpp-1.9.5/dist/jsoncpp.cpp -I jsoncpp-1.9.5/dist
```

**Comment by BillyDonahue** [comment]
i don't think so. That's not a Makefile.
You need to pick a build system like make or CMake or something and do what they do

**Comment by XY-ShadowK** [comment]
> i don't think so. That's not a Makefile.

I know, it is just for debugging at development time, when you haven't added a task to the new source file.
I want to know whether it need more conditional judgment or environment variable setting to build cross-platform.
```makefile
if(WIN32)
    set(CMAKE_CXX_COMPILER "MSVC")
    set(MIN_SDK_VERSION "10.0.19041")
    set(MIN_MSVC_VERSION "1920")
elseif(UNIX)
    set(CMAKE_CXX_COMPILER "g++")
endif()
```

**Comment by BillyDonahue** [comment]
i don't know those details. sorry. But yeah if all you have is a shell script with one g++ command line then just add the amalgamated jsoncpp.cpp to that.

But we're outside of jsoncpp issues here.

**Comment by XY-ShadowK** [comment]
Thanks for you patience.
You're the gentlest developer I've ever seen on github.
